axelor-base/src/main/resources/views/Menu.xml

```diff
@@ -530,6 +530,22 @@
 	    <view type="form" name="company-department-form"/>
 	</action-view>
 
+	<menuitem name="company-bank-details" parent="admin-root-user-configuration" title="Bank details" action="company.bank.details"/>
+
+	<action-view name="company.bank.details" title="Company.bankDetails" model="com.axelor.apps.base.db.BankDetails">
+    	<view type="cards" name="bank-details-cards"/>
+    	<view type="grid" name="bank-details-grid"/>
+	    <view type="form" name="bank-details-form"/>
+	    <domain>self.id IN (:bankDetailsList)</domain>
+	    <context name="bankDetailsList" expr="eval: __repo__(Company).all().fetch().collect{ it.getBankDetailsList().collect { it.getId() } }.flatten()"/>
```

Comment:
Instead of finding company and getting ids of bank details, use reverse condition, find all bank details where company is not null.